src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -566,6 +566,8 @@ seastar::future<>
 ExtentPlacementManager::BackgroundProcess::do_background_cycle()
 {
   assert(is_ready());
+  maybe_update_eviction_mode();
```

Comment:
Is it better to be called in `Cache::complete_commit()` instead?

Reply:
Looks reasonable, but `maybe_update_eviction_mode` becomes a public interface of EPM.

Reply:
It's reasonable to me that the EPM state needs to be updated after each commit. Probably just name it `EPM::complete_commit()`?

Reply:
@zhscn I'm still wondering where we should update the eviction mode in order to make sure it is always up-to-date.

Now it seems to me that `background_should_run()` is the only place to do the update.

Reply:
> we should update the eviction mode in order to make sure it is always up-to-date.

Why should we keep it up-to-data? It looks to me that the `eviction_state_t` only affects the rewrite background transction(trim and clean).

So it is safe and sufficient to update eviction mode before starting the background process, so I place it at the start of `BackgroundProcess::do_background_cycle`.

Reply:
I think keep it up-to-date is helpful to adjust `INIT_GENERATION` extents from client IO, which looks similar to write through policy. But it seems not a good idea to handle this case in eviction process, or am I miss something?

Reply:
The reason is from the logic in this PR: https://github.com/ceph/ceph/blob/54e997790bdf5acb96998c4de0a1ad71657ef5aa/src/crimson/os/seastore/extent_placement_manager.h#L880-L883

I think it implies that eviction_mode may not be up-to-date with the `main_alive_ratio`.

This is why I suggest to move it to `background_should_run()` as it is atomically called before `do_background_cycle()`: https://github.com/ceph/ceph/blob/54e997790bdf5acb96998c4de0a1ad71657ef5aa/src/crimson/os/seastore/extent_placement_manager.cc#L550-L552